angular/src/app/app.component.ts

```diff
@@ -1,26 +1,89 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+
+import { PlayerService } from './player.service';
+import { Player } from './player';
+
 
 @Component({
   selector: 'my-app',
+  styles: [`
+  .selected {
+    background-color: #CFD8DC !important;
+    color: white;
+  }
+  .players{
+    margin: 0 0 2em 0;
+    list-style-type: none;
+    padding: 0;
+    width: 15em;
+  }
+  .players li {
+    cursor: pointer;
+    position: relative;
+    left: 0;
+    background-color: #EEE;
+    margin: .5em;
+    padding: .3em 0;
+    height: 1.6em;
+    border-radius: 4px;
+  }
+  .players li.selected:hover {
+    background-color: #BBD8DC !important;
+    color: white;
+  }
+  .players li:hover {
+    color: #607D8B;
+    background-color: #DDD;
+    left: .1em;
+  }
+  .players .text {
+   position: relative;
+   top: -3px;
+  }
+  .players .badge {
+    display: inline-block;
+    font-size: small;
+    color: white;
+    padding: 0.8em 0.7em 0 0.7em;
+    background-color: #607D8B;
+    line-height: 1em;
+    position: relative;
+    left: -1px;
+    top: -4px;
+    height: 1.8em;
+    margin-right: .8em;
+    border-radius: 4px 0 0 4px;
+  }
+  `],
   template: `
     <h1>{{title}}</h1>
-    <h2>{{player.name}}</h2>
-    <div><label>ldap: </label>{{player.ldap}}</div>
-    <div>
-      <label>name: </label>
-      <input [(ngModel)]="player.name" placeholder="name">
-    </div>
+    <ul class="players">
+      <li *ngFor="let player of players"
+       [class.selected]="player === selectedPlayer"
+       (click)="onSelect(player)">
```

Comment:
Ach! This is just because I haven't told vim how to indent these files properly. Thanks for the catch. To be clear, are you suggesting fixing my one-space indent to a two-space indent, or are you specifically suggesting going to four spaces?

> The angular tutorial actually has those continuation lines indented by the same amount as the enclosed `<span>` tag, which seems like the worst option to me...

Oh I dunno, I bet we can do worse...

```html
    <li *ngFor="let player of players"
  [class.selected]="player === selectedPlayer"
  (click)="onSelect(player)">
    <span class="badge">{{player.ldap}}</span> {{player.name}}
  </li>
```